builder/shell-for.nix

```diff
@@ -74,14 +75,23 @@ let
     # inherit (hsPkgs) hoogle;
   };
 
-  mkDrvArgs = builtins.removeAttrs args ["packages" "additional" "withHoogle"];
+  mkDrvArgs = builtins.removeAttrs args ["packages" "additional" "withHoogle" "tools"];
 in
   stdenv.mkDerivation (mkDrvArgs // {
     name = mkDrvArgs.name or name;
 
     buildInputs = systemInputs
       ++ mkDrvArgs.buildInputs or []
-      ++ lib.optional withHoogle' hoogleIndex;
+      ++ lib.optional withHoogle' hoogleIndex
+      ++ lib.mapAttrsToList (name: versionOrArgs:
```

Comment:
I think this could use a comment. It's not obvious what we are doing here without staring at the code for a while.

Reply:
I have refactored this a and included some comments.  Also switched to nativeBuildInputs as per @michaelpj's suggestion [here](https://github.com/input-output-hk/haskell.nix/pull/560#discussion_r414581645).